projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -228,31 +224,27 @@ internal static InboundFrame ReadFrom(Stream reader, byte[] frameHeaderBuffer)
                     throw new EndOfStreamException("Reached the end of the stream. Possible authentication failure.");
                 case 'A':
                     // Probably an AMQP protocol header, otherwise meaningless
-                    ProcessProtocolHeader(reader);
+                    await ProcessProtocolHeader(reader, frameHeaderBuffer.Slice(1, 6)).ConfigureAwait(false);
```

Comment:
why not Slice(1, 7) and avoid the async code in ProcessProtocolHeader

Reply:
this swich could also use a default case that throws because of protocol violation?!

Reply:
I might then read more than I want and complicate the code that comes later. Also, this is an edge case :)